processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -260,13 +248,28 @@ func (c *WatchClient) Start() error {
 		go c.nodeInformer.Run(c.stopCh)
 	}
 
+	reg, err = c.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.handlePodAdd,
+		UpdateFunc: c.handlePodUpdate,
+		DeleteFunc: c.handlePodDelete,
+	})
+	if err != nil {
+		return err
+	}
+
```

Comment:
Why not add podInformer's HasSynced into the slice `synced`

Reply:
the `synced` slice contains the informers that need to be completed before the pod informer can start. Adding the pod informer's HasSynced to this slice would therefore cause the informer to stall indefinitely. Once the pod informer has started, we wait for it to be finished further down below - i.e. [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37138/files#diff-da945276e27b1e0a3eed6a7e4a97c5b1bb9d90da43bd8ef9fce8b533584d4292R272)

Reply:
Got it, thanks for explaining.